index.coffee

```diff
@@ -20,21 +20,24 @@ tryExtractDigestHash = (evt) ->
 		matchPull = evt.status.match(/^Digest:\s([a-zA-Z0-9]+:[a-f0-9]+)$/)
 		return matchPull[1] if matchPull?
 
-awaitRegistryStream = (stream, onProgress) ->
+awaitRegistryStream = (stream, onProgress, ignoreErrorEvents) ->
 	contentHash = null
 	new Promise (resolve, reject) ->
 
 		jsonStream = JSONStream.parse()
 
 		jsonStream.on 'data', (evt) ->
 			if typeof evt isnt 'object'
-				evt = {}
-			else
+				return
+			try
+				if evt.error and !ignoreErrorEvents
+					throw new Error(evt.error)
```

Comment:
A sample `evt` object that triggers this test is:
```
{ errorDetail: { message: 'unauthorized: authentication required' },
  error: 'unauthorized: authentication required' }
```

Reply:
Great, this has been a major thorn in my side with `deploy` in the past!